Dockerfile

```diff
@@ -1,66 +1,97 @@
+FROM ubuntu:18.04 AS builder
+
+LABEL maintainer="team@appwrite.io"
+
+ARG TESTING=false
+
+ENV TZ=Asia/Tel_Aviv \
+    DEBIAN_FRONTEND=noninteractive \
+    PHP_VERSION=7.3 \
+    PHP_REDIS_VERSION=3.1.2
+
+RUN \
+  apt-get update && \
+  apt-get install -y --no-install-recommends --no-install-suggests ca-certificates software-properties-common wget curl git openssl && \
+  LC_ALL=C.UTF-8 add-apt-repository -y ppa:ondrej/php && \
+  apt-get update && \
+  apt-get install -y --no-install-recommends --no-install-suggests make php$PHP_VERSION php$PHP_VERSION-dev zip unzip php$PHP_VERSION-zip && \
+  # redis php extension
+  wget -q https://github.com/phpredis/phpredis/archive/$PHP_REDIS_VERSION.tar.gz && \
+  tar -xf $PHP_REDIS_VERSION.tar.gz && \
+  cd phpredis-$PHP_REDIS_VERSION && \
+  phpize$PHP_VERSION && \
+  ./configure && \
+  make && \
+  # composer
+  curl -sS https://getcomposer.org/installer | php -- --install-dir=/usr/bin --filename=composer
+
+WORKDIR /usr/local/src/
+# Updating PHP dependencies and auto-loading...
+ENV TESTING=$TESTING
+COPY composer.* /usr/local/src/
+RUN composer update --ignore-platform-reqs --optimize-autoloader \
+    --no-plugins --no-scripts --prefer-dist \
+    `if [ "$TESTING" != "true" ]; then echo "--no-dev"; fi`
+
```

Comment:
We can skip this and just use production settings. All current tests are actually e2e so they don't have to be part of the container.

Reply:
For this comment and other comments below that are about the usage of `TESTING=false/true`, all of these modifications related to the running tests inside the container.
If we are going to run them outside then we need to prepare the environment itself to run the tests

Reply:
Related comment: https://github.com/appwrite/appwrite/pull/109#discussion_r349878947